fix(oauth): token.json-Persistenz scheitert unter Windows mit WinError 5 (os.replace, intermittierend)

## Symptom

Beim Sync-Pull scheitert die Token-Persistenz unter Windows mit `PermissionError [WinError 5] Zugriff verweigert` — der komplette Pull bricht ab („Sync pull failed"). Beobachtet beim lokalen Lauf der installierten App:

```
[ERROR] __main__: Sync pull failed
Traceback (most recent call last):
  File "main.py", line 67, in _run_pull_in_background
  File "src\drive.py", line 99, in get_drive_service
  File "src\oauth_utils.py", line 40, in write_token
PermissionError: [WinError 5] Zugriff verweigert:
  '%LOCALAPPDATA%\Programs\Zeiterfassung\.token-XXXX.tmp'
  -> '%LOCALAPPDATA%\Programs\Zeiterfassung\token.json'
```

## Ursache

`write_token` (`src/oauth_utils.py:40`) schreibt atomar via `tempfile.mkstemp` + `os.replace(tmp, token.json)`. Auf **Windows** ist `os.replace` nicht so robust wie unter POSIX: Hält irgendein Prozess ein offenes Handle auf `token.json` **oder** scannt der Defender-Echtzeitschutz gerade die frisch erzeugte `.token-*.tmp`, verweigert das OS den Rename mit `WinError 5`. Das Verzeichnis liegt bei der Windows-Installation direkt neben der Exe (`%LOCALAPPDATA%\Programs\Zeiterfassung\`), wo Defender aktiv scannt.

Der Fehler ist **intermittierend** — beim direkt darauffolgenden App-Neustart trat er nicht auf. Trotzdem: tritt er auf, geht der Token-Refresh verloren und der Sync-Pull schlägt komplett fehl.

## Vorschlag

Kurze Retry-Schleife um `os.replace` mit Backoff, die genau `PermissionError`/`OSError` (WinError 5) abfängt — analog zum Retry-Muster in #117 (holidays `FileNotFoundError`). Nach N erfolglosen Versuchen sauber durchreichen. So überbrückt man den kurzen AV-/Handle-Lock, ohne die Atomarität aufzugeben.

## Kontext

Gefunden beim lokalen Test-Build des gemergten Audit-`master` (`1.17.0-dev`, alle 12 Audit-PRs). **Vorbestehend**, keine Regression durch die Merges (steht schon im Log vor dem Update). Betrifft nur Windows.

--- a/src/oauth_utils.py
+++ b/src/oauth_utils.py
@@ -13,6 +13,7 @@ import json
 import os
 import stat
 import tempfile
+import time
 
 
 def write_token(creds, token_path):
@@ -37,7 +38,22 @@ def write_token(creds, token_path):
             os.chmod(tmp_path, stat.S_IRUSR | stat.S_IWUSR)  # 0o600
         except OSError:
             pass
-        os.replace(tmp_path, token_path)
+        # os.replace mit Retry gegen transiente Windows-PermissionError: ein
+        # Virenscanner, der die frisch erzeugte .token-*.tmp scannt, oder ein
+        # noch offenes Handle auf token.json blockiert den atomaren Rename kurz
+        # (WinError 5/32 -> beide PermissionError). Kurzer Backoff überbrückt das;
+        # bleibt es dabei, wird der Fehler durchgereicht (Issue #135, Muster wie
+        # #117). Gezielt PermissionError, damit echte Fehler (fehlende tmp,
+        # Zielverzeichnis) nicht maskiert werden.
+        attempts = 5
+        for attempt in range(attempts):
+            try:
+                os.replace(tmp_path, token_path)
+                break
+            except PermissionError:
+                if attempt == attempts - 1:
+                    raise
+                time.sleep(0.2)
     except BaseException:
         try:
             os.remove(tmp_path)

--- a/tests/test_oauth_utils.py
+++ b/tests/test_oauth_utils.py
@@ -8,6 +8,8 @@ import os
 import stat
 from unittest.mock import MagicMock
 
+import pytest
+
 from src.oauth_utils import write_token, discard_token_for_scope_upgrade
 
 
@@ -58,6 +60,56 @@ def test_write_token_overwrites_existing(tmp_path):
 
     with open(path) as f:
         assert f.read() == "new"
+
+
+def test_write_token_retries_transient_permission_error(tmp_path, monkeypatch):
+    """Regression #135: ein transienter Windows-PermissionError (AV-Scan/offenes
+    Handle blockiert os.replace, WinError 5/32) wird per Retry überbrückt — der
+    Token landet trotzdem sauber auf der Platte."""
+    path = str(tmp_path / "token.json")
+    creds = MagicMock()
+    creds.to_json.return_value = '{"token": "abc"}'
+
+    real_replace = os.replace
+    calls = {"n": 0}
+
+    def flaky_replace(src, dst):
+        calls["n"] += 1
+        if calls["n"] == 1:
+            raise PermissionError(13, "Zugriff verweigert")
+        return real_replace(src, dst)
+
+    monkeypatch.setattr("src.oauth_utils.os.replace", flaky_replace)
+    monkeypatch.setattr("time.sleep", lambda _s: None)
+
+    write_token(creds, path)
+
+    assert calls["n"] == 2  # ein Fehlschlag, dann Erfolg
+    with open(path) as f:
+        assert f.read() == '{"token": "abc"}'
+
+
+def test_write_token_reraises_persistent_permission_error(tmp_path, monkeypatch):
+    """Bleibt der PermissionError dauerhaft, werden alle Versuche ausgeschöpft
+    und der Fehler durchgereicht (nicht still verschluckt); keine Temp-Reste."""
+    path = str(tmp_path / "token.json")
+    creds = MagicMock()
+    creds.to_json.return_value = "{}"
+
+    calls = {"n": 0}
+
+    def always_fail(src, dst):
+        calls["n"] += 1
+        raise PermissionError(13, "Zugriff verweigert")
+
+    monkeypatch.setattr("src.oauth_utils.os.replace", always_fail)
+    monkeypatch.setattr("time.sleep", lambda _s: None)
+
+    with pytest.raises(PermissionError):
+        write_token(creds, path)
+
+    assert calls["n"] > 1  # es wurde wiederholt, nicht nur einmal versucht
+    assert os.listdir(str(tmp_path)) == []  # tmp aufgeräumt, kein token.json
 
 
 def _write_token_file(path, scopes):

